mautic/mautic
# add "including one of" segment for multiselect fields

**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | 
| New feature? | 
| Automated tests included? |
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | 
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
This PR added support for **include one of** expression for multi select field.
By default include/exclude expect selected values in condition.
**Include one of** expect at least on of selected value.

For example: 
- include - if we expect fruit in filter - orange, lemon - contact must these values in profile
- **Include one of**- If we expect fruit in filter - orange, lemon - contact could have at least one of them in profile

![image](https://user-images.githubusercontent.com/462477/69881729-2e71e300-12ce-11ea-80dd-c8e1326f8dc4.png)

#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com). On  your own test enviroment require assets generate **php app/console m:a:g**
2. Create multiselect contact field  with multiple options
3. Create segment with filter **Include one of** some values  of that multiselect field
4. Create contact with one of that values
5. Segment rebuild
6. Your contact should build in your segment
7. If you change expression from **Include one of** to include, you contact should not be passed by segment filter  and should remove from it
